src/Microsoft.Data/PrimitiveDataFrameColumn.cs

```diff
@@ -170,13 +170,19 @@ protected internal override Apache.Arrow.Array ToArrowArray(long startIndex, int
             }
         }
 
-        protected override object GetValue(long startIndex, int length)
+        protected override List<object> GetValues(long startIndex, int length)
         {
             if (startIndex > Length)
             {
                 throw new ArgumentOutOfRangeException(nameof(startIndex));
             }
-            return _columnContainer[startIndex, length];
+            var ret = new List<object>(length);
+            long endIndex = Math.Min(Length, startIndex + length);
+            for (long i = startIndex; i < endIndex; i++)
+            {
+                ret.Add(this[i]);
```

Comment:
I'm not certain this is an improvement. With this new change, you are going to box every value when adding it to this list. And then callers will need to unbox the values to get the underlying `T` values.

Reply:
It's definitely not an improvement. I was forced to do this because I can't return an `IList<T?>`(returned by `PrimitiveColumnContainer`) as an `IList<object>` directly. However, I can instead return this as `return _columnContainer[startIndex, length] as dynamic` which is returning `IList<T?> -> IList<object>`. That would result in no boxing when the list is being returned. Know of a better/less ugly way?

Reply:
Returning a IReadOnlyList<T?> may work here. https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/covariance-contravariance/variance-in-generic-interfaces

Reply:
Sigh, returning `IReadOnlyList<T?>` doesn't work, because variance in generic interfaces is restricted to reference types and `T` here is unmanaged constrained

Reply:
Maybe, let's take a step back. What is the value of this API? How do we expect users to use it?

Reply:
This is meant to be a weakly typed way to get _n_ rows in a `DataFrameColumn`. So, something like this to return the values between indices 20 and 29:
`var column = dataFrame["SomeColumnName"]
column[20, 10]` 
Without this method, there'd be no way to get more than 1 row at a time from the base `DataFrameColumn`. On the derived types, the indexers are strongly typed, so this is not an issue.